docs/source/library-user-guide/extending-sql.md

```diff
@@ -0,0 +1,339 @@
+
+| Method                      | Purpose                                         |
+| --------------------------- | ----------------------------------------------- |
+| `plan(relation)`            | Recursively plan a nested relation              |
+| `sql_to_expr(expr, schema)` | Convert SQL expression to DataFusion Expr       |
+| `context_provider()`        | Access session configuration, tables, functions |
+
+See the [RelationPlanner API documentation] for additional methods like
+`normalize_ident()` and `object_name_to_table_reference()`.
+
+#### Implementation Strategies
+
+There are two main approaches when implementing a `RelationPlanner`:
+
+1. **Rewrite to Standard SQL**: Transform custom syntax into equivalent standard
+   operations that DataFusion already knows how to execute (e.g., PIVOT → GROUP BY
+   with CASE expressions). This is the simplest approach when possible.
+
+2. **Custom Logical and Physical Nodes**: Create a `UserDefinedLogicalNode` to
+   represent the operation in the logical plan, along with a custom `ExecutionPlan`
+   to execute it. Both are required for end-to-end execution.
+
+#### Example: Basic RelationPlanner Structure
+
+```rust
+use std::sync::Arc;
+use datafusion::error::Result;
+use datafusion::prelude::*;
+use datafusion_expr::planner::{
+    PlannedRelation, RelationPlanner, RelationPlannerContext, RelationPlanning,
+};
+use datafusion_sql::sqlparser::ast::TableFactor;
+
+#[derive(Debug)]
+struct MyRelationPlanner;
+
+impl RelationPlanner for MyRelationPlanner {
+    fn plan_relation(
+        &self,
+        relation: TableFactor,
+        ctx: &mut dyn RelationPlannerContext,
+    ) -> Result<RelationPlanning> {
+        match relation {
+            // Handle your custom relation
+            TableFactor::Pivot { table, alias, .. } => {
+                // Plan the input table
+                let input = ctx.plan(*table)?;
+
+                // Transform or wrap the plan as needed
+                // ...
+
+                Ok(RelationPlanning::Planned(PlannedRelation::new(input, alias)))
+            }
+
+            // Return Original for relations you don't handle
+            other => Ok(RelationPlanning::Original(other)),
+        }
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let ctx = SessionContext::new();
+
+    // Register the custom planner
+    ctx.register_relation_planner(Arc::new(MyRelationPlanner))?;
+
+    Ok(())
+}
+```
+
+## Complete Examples
+
+The DataFusion repository includes comprehensive examples demonstrating each
+approach:
+
+### TABLESAMPLE (Custom Logical and Physical Nodes)
+
+The [table_sample.rs] example shows a complete end-to-end implementation including:
+
+- Parsing TABLESAMPLE syntax via `RelationPlanner`
+- Custom logical node (`TableSamplePlanNode`)
+- Custom physical operator (`SampleExec`)
+- Bernoulli sampling with reproducible seeds
+
+```sql
+SELECT * FROM table TABLESAMPLE BERNOULLI(10 PERCENT) REPEATABLE(42)
+```
```

Comment:
I think some of this detail is unnecessary as it repeats what is already in the examples -- I think you could remove the bullet points and this would be just as good. Something like

(the same comment applies to the others too)

```suggestion
The [table_sample.rs] example shows a complete end-to-end implementation of how to support queries such as

```sql
SELECT * FROM table TABLESAMPLE BERNOULLI(10 PERCENT) REPEATABLE(42)
```
```